content/docs/command-reference/exp/apply.md

```diff
@@ -15,21 +15,25 @@ positional arguments:
 ## Description
 
 Restores an `experiment` into the workspace as long as no more Git commits have
-been made after the target experiment (`HEAD` hasn't moved). The `experiment`
-can be referenced by name or hash (see `dvc exp run` for details). This changes
-any files (code, data, <abbr>parameters</abbr>, <abbr>metrics</abbr>, etc.)
-needed to reflect the experiment conditions and results in the workspace.
+been made after the target experiment (`HEAD` hasn't moved). The experiment can
+be referenced by name or hash (see `dvc exp run` for details).
 
-⚠️ Conflicting changes in the workspace are overwritten unless `--no-force` is
-used.
+Specifically, `dvc exp apply` changes any files (code, data,
+<abbr>parameters</abbr>, <abbr>metrics</abbr>, etc.) needed to reflect the
+experiment conditions and results in the workspace. Current changes to the
+workspace are preserved except if they conflict with the experiment in question.
```

Comment:
I think "current changes" is not clear, and may be confusing. I think the next ⚠️ text is clear, no need for this sentence.

Reply:
Idk @iesahin ... I think it's valuable to explicitly state that unstaged changes can end up mixed in with the applied experiment. Users are responsible to only add/commit the relevant changes at that point. I reworked this whole intro now, PTAL.

Reply:
Wait. This is not true though! `exp apply` actually destroys all other changes (I just checked)... ⌛

Reply:
OK I updated this with what I think is the desired current behavior, although it's not working properly for me (opened https://github.com/iterative/dvc/issues/7180).